File: solutions/prolog/triangle/1/triangle.pl
validate_triagle_sides(S1, S2, S3) :- 
    S1 =< 0; 
    S2 =< 0; 
    S3 =< 0; 
    S1 + S2 < S3;
    S1 + S3 < S2;
    S2 + S3 < S1.

triangle(S1, S2, S3, _) :- validate_triagle_sides(S1, S2, S3), !, fail.
triangle(S, S, S, Type) :- Type = "equilateral".
triangle(S, S, _, Type) :- Type = "isosceles".
triangle(S, _, S, Type) :- Type = "isosceles".
triangle(_, S, S, Type) :- Type = "isosceles".
triangle(S1, S2, S3, Type) :- S1 =\= S2, S1 =\= S3, S2 =\= S3, Type = "scalene".

File: solutions/php/triangle/2/Triangle.php
<?php

declare(strict_types=1);

class Triangle
{
    public function __construct
        (private float $a, private float $b, private float $c)
    {
        if ($a <= 0 || $b <= 0 || $c <= 0) {
            throw new Exception("All sides must be greater than zero");
        }
        if ($c >= $a + $b || $b >= $a + $c || $a >= $b + $c) {
            throw new Exception(
                "The biggest side must be smaller " .
                    "than the sum of the other 2 sides");
        }
    }

    public function kind(): string
    {
        if ($this->a == $this->b && $this->a == $this->c) {
            $kind = 'equilateral';
        } else if ($this->a == $this->b
                   || $this->a == $this->c || $this->b == $this->c) {
            $kind = 'isosceles';
        }else {
            $kind = 'scalene';
        }
        return $kind;
    }
}
